cpp/src/traversal/sssp.cu

```diff
@@ -314,91 +310,42 @@ void sssp(Graph* graph,
-      ALLOC_TRY((void**)&d_edge_data, edge_data_size, stream);
-      CUDA_TRY(cudaMemcpy(d_edge_data,
-                          &h_edge_data[0],
-                          edge_data_size,
-                          cudaMemcpyHostToDevice));
-      gdf_column_view(graph->adjList->edge_data,
-                      d_edge_data,
-                      nullptr,
-                      graph->adjList->indices->size,
-                      GDF_FLOAT64);
-
-    } else {
-      // Else generate float
-      std::vector<float> h_edge_data(graph->adjList->indices->size, 1.0);
-      size_t edge_data_size = sizeof(float) * h_edge_data.size();
-      ALLOC_TRY((void**)&d_edge_data, edge_data_size, stream);
-      CUDA_TRY(cudaMemcpy(d_edge_data,
-                          &h_edge_data[0],
-                          edge_data_size,
-                          cudaMemcpyHostToDevice));
-      gdf_column_view(graph->adjList->edge_data,
-                      d_edge_data,
-                      nullptr,
-                      graph->adjList->indices->size,
-                      GDF_FLOAT32);
-    }
+    std::vector<WT> h_edge_data(graph.number_of_edges, static_cast<WT>(1));
+    size_t edge_data_size = sizeof(WT) * h_edge_data.size();
+    ALLOC_TRY((WT**)&edge_weights_ptr, edge_data_size, stream);
+    CUDA_TRY(cudaMemcpy(edge_weights_ptr,
+                        &h_edge_data[0],
+                        edge_data_size,
+                        cudaMemcpyHostToDevice));
+    should_free_weights = true;
   } else {
-    // Got weighted graph
-    CUGRAPH_EXPECTS(
-        graph->adjList->edge_data->size == graph->adjList->indices->size,
-        "Graph sizes mismatch");
-    // TODO fix me after gdf_column is removed from Graph
-    CUGRAPH_EXPECTS(graph->adjList->edge_data->dtype == GDF_FLOAT32 ||
-                    graph->adjList->edge_data->dtype == GDF_FLOAT64,
-                "Invalid API parameter");
-    // TODO fix me after gdf_column is removed from Graph
-    // if (distances) CUGRAPH_EXPECTS(typeid(distances) == typeid(graph->adjList->edge_data), "distances and weights type mismatch");
-
     // SSSP is not defined for graphs with negative weight cycles
     // Warn user about any negative edges
-    if (graph->prop && graph->prop->has_negative_edges == GDF_PROP_TRUE)
+    if (graph.prop.has_negative_edges == experimental::PropType::PROP_TRUE)
       std::cerr << "WARN: The graph has negative weight edges. SSSP will not "
                    "converge if the graph has negative weight cycles\n";
+    edge_weights_ptr = (WT*)graph.edge_data;
   }
 
-  int n = graph->adjList->offsets->size - 1;
-  int e = graph->adjList->indices->size;
-  int* offsets_ptr = (int*)graph->adjList->offsets->data;
-  int* indices_ptr = (int*)graph->adjList->indices->data;
+  int n = graph.number_of_vertices;
+  int e = graph.number_of_edges;
 
-  void* edge_weights_ptr = static_cast<void*>(graph->adjList->edge_data->data);
+  VT* offsets_ptr = (VT*)graph.offsets;
```

Comment:
```suggestion
  ET* offsets_ptr = graph.offsets;
```
Should be `ET*`?
Is cast needed?

Reply:
Updated in https://github.com/rapidsai/cugraph/pull/764/commits/1528aebe9ea39d6ac0060b1068a76f8c2a36056c, changed `VT*` to `const ET*`